SECEdgar/network_client.py

```diff
@@ -0,0 +1,131 @@
+import requests
+import time
+
+from SECEdgar.utils.exceptions import EDGARQueryError
+
+
+class NetworkClient(object):
```

Comment:
This is great but we can improve a little bit here.

Say tomorrow, I want to switch to different network client; then we will have to change all the implementation details in so many files and so we will use `ABC` to decouple this dependency. (Dependency Inversion Principle)

A network client should just have one method or multiple method depending how you want expose the API. For us, if we provide the request object then just load method will be enough.

Later when we switch to async request we can add another method with async callback.


```
class SECEdgarService(object):
   # here request object will contain all the info about the URL request like whether it is get or post, what are the header etc.
    def load(self, request):
        raise NotImplementedError('subclass should override')

class SECEdgarClient(SECEdgarService):
    def load(self, request):
       // map the request object to particular lib object.
       // fetch the request from whatever lib like request or httpx etc
        return result
```

Hope this helps. Unless we can collaborate on a doc.

Reply:
I am not exactly sure what you mean here. Does this seem like something we should be able to put off at least for v0.1.5?

Reply:
Sure we can put it for `v0.1.5`.

Reply:
Putting an example here to help you what I was trying to say.

```
interface NetworkService {
   get(url)
   save(data)
}

class SECEdgar implements NetworkService {
   get(url) {
       requests.get(url)
   }
   save(data) {
      
   }
}

class MockSECEdgar implements NetworkService {
   get(url) {
       load save saved html file from disk and return it
   }
   save(data) {
      just print it or do whatever
   }
}
```

Reply:
It is Java'ish code but just trying to explain the idea behind it.